service/controller/app/v1/resource_set.go

```diff
@@ -159,12 +158,17 @@ func NewResourceSet(config ResourceSetConfig) (*controller.ResourceSet, error) {
 			return nil, microerror.Mask(err)
 		}
 
-		g8sClient, err := kubeConfig.NewG8sClientForApp(ctx, cr)
+		restConfig, err := kubeConfig.NewRESTConfigForApp(ctx, cr)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
```

Comment:
This is the actual change. We get the REST config once and then add the clients to the context.